tests/devices/default_qubit/test_default_qubit.py

```diff
@@ -1637,6 +1638,272 @@ def test_projector_dynamic_type(max_workers, n_wires):
+            (qml.var(qml.PauliZ(0)), "autograd", True),
+            (qml.probs(wires=[0, 1]), "autograd", True),
+            (qml.state(), "autograd", True),
+            (qml.density_matrix(wires=0), "autograd", True),
+            (qml.purity(0), "numpy", True),
+            (qml.vn_entropy(0), "numpy", False),
+            (qml.mutual_info(0, 1), "numpy", False),
+        ],
+    )
+    def test_postselection_invalid_analytic(
+        self, mp, autograd_interface, is_nan, interface, use_jit
+    ):
+        """Test that the results of a qnode are nan values of the correct shape if the state
+        that we are postselecting has a zero probability of occurring."""
+
+        if (isinstance(mp, qml.measurements.MutualInfoMP) and interface != "jax") or (
+            isinstance(mp, qml.measurements.VnEntropyMP) and interface == "tensorflow"
+        ):
+            pytest.skip("Unsupported measurements and interfaces.")
+
+        if use_jit and interface != "jax":
+            pytest.skip("Can't jit with non-jax interfaces.")
+
+        # Wires are specified so that the shape for measurements can be determined correctly
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface=interface)
+        def circ():
+            qml.RX(np.pi, 0)
+            qml.CNOT([0, 1])
+            qml.measure(0, postselect=0)
+            return qml.apply(mp)
+
+        if use_jit:
+            import jax
+
+            circ = jax.jit(circ)
+
+        res = circ()
+        if interface == "autograd":
+            assert qml.math.get_interface(res) == autograd_interface
+        else:
+            assert qml.math.get_interface(res) == interface
+        assert qml.math.shape(res) == mp.shape(dev, qml.measurements.Shots(None))
+        if is_nan:
+            assert qml.math.all(qml.math.isnan(res))
+        else:
+            assert qml.math.allclose(res, 0.0)
+
+    @pytest.mark.parametrize(
+        "mp, expected_shape",
+        [
+            (qml.expval(qml.PauliZ(0)), ()),
+            (qml.var(qml.PauliZ(0)), ()),
+            (qml.sample(qml.PauliZ(0)), (0,)),
+            (qml.classical_shadow(wires=0), (2, 0, 1)),
+            (qml.shadow_expval(qml.Hamiltonian([1, 1], [qml.PauliZ(0), qml.PauliX(0)])), ()),
+            # qml.probs and qml.counts are not tested because they fail in this case
+        ],
+    )
+    @pytest.mark.parametrize("shots", [10, (10, 10)])
+    def test_postselection_invalid_finite_shots(
+        self, mp, expected_shape, shots, interface, use_jit
+    ):
+        """Test that the results of a qnode are nan values of the correct shape if the state
+        that we are postselecting has a zero probability of occurring with finite shots."""
+
+        if use_jit and interface != "jax":
+            pytest.skip("Can't jit with non-jax interfaces.")
+
+        # Wires are specified so that the shape for measurements can be determined correctly
```

Comment:
what happens if wires are not specified?

Reply:
`StateMP.shape` needs device wires to determine shape, so the tests for `qml.state()` fail.